modules/apps/29-fee/keeper/msg_server.go

```diff
@@ -25,9 +27,51 @@ func (k Keeper) RegisterCounterpartyAddress(goCtx context.Context, msg *types.Ms
 	return &types.MsgRegisterCounterpartyAddressResponse{}, nil
 }
 
-// EscrowPacketFee defines a rpc handler method for MsgEscrowPacketFee
-// EscrowPacketFee is an open callback that may be called by any module/user that wishes to escrow funds in order to
-// incentivize the relaying of the given packet.
-func (k Keeper) EscrowPacketFee(goCtx context.Context, msg *types.MsgEscrowPacketFee) (*types.MsgEscrowPacketFeeResponse, error) {
-	return &types.MsgEscrowPacketFeeResponse{}, nil
+// PayPacketFee defines a rpc handler method for MsgPayPacketFee
+// PayPacketFee is an open callback that may be called by any module/user that wishes to escrow funds in order to relay the packet with the next sequence
+func (k Keeper) PayPacketFee(goCtx context.Context, msg *types.MsgPayPacketFee) (*types.MsgPayPacketFeeResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// get the next sequence
+	sequence, found := k.GetNextSequenceSend(ctx, msg.SourcePortId, msg.SourceChannelId)
+	if !found {
+		return nil, channeltypes.ErrSequenceSendNotFound
+	}
+
+	packetId := channeltypes.PacketId{
+		PortId:    msg.SourcePortId,
+		ChannelId: msg.SourceChannelId,
+		Sequence:  sequence,
+	}
+
+	refundAccAddr, err := sdk.AccAddressFromBech32(msg.RefundAccount)
+	if err != nil {
+		return &types.MsgPayPacketFeeResponse{}, err
+	}
+
+	err = k.escrowPacketFee(ctx, refundAccAddr, *msg.Fee, packetId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.MsgPayPacketFeeResponse{}, nil
+}
+
+// PayPacketFee defines a rpc handler method for MsgPayPacketFee
+// PayPacketFee is an open callback that may be called by any module/user that wishes to escrow funds in order to
+// incentivize the relaying of a known packet
+func (k Keeper) PayPacketFeeAsync(goCtx context.Context, msg *types.MsgPayPacketFeeAsync) (*types.MsgPayPacketFeeAsyncResponse, error) {
```

Comment:
@colin-axner @AdityaSripal Although the internal logic between `PayPacket` & `PayPacketAsync` is not massively different, I think from an API perspective it makes sense to keep them separate. As far as the user is concerned it is very different functionality. 

In the first case, the user is bundling messages together in a single packet to pay for the packet at the next sequence. In the second case, the user wants to pay for a packet that has been committed already. 

wdyt?

Reply:
The public facing RPCs should definitely be different. but they can both call the same helper function